analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocations.cs

```diff
@@ -0,0 +1,216 @@
+            validator.ValidateTag("BeforeExtensionNull", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+            validator.ValidateTag("BeforePreserve", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
+            validator.ValidateTag("AfterInstance", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue("Instance method should set NotNull constraint."));
+            validator.ValidateTag("AfterExtensionArg", x => x.Should().BeNull("Extensions can run on null instances."));
+            validator.ValidateTag("AfterExtensionNull", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue("Extensions can run on null instances."));
+            validator.ValidateTag("AfterPreserve", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue("Other constraints should not be removed."));
+        }
+
+        [TestMethod]
+        public void Invocation_SetsNotNullOnInstance_VB()
+        {
+            const string code = @"
+Public Class Sample
+
+    Public Sub Main(InstanceArg As Sample, StaticArg As Sample, ExtensionArg As Sample)
+        Tag(""BeforeInstance"", InstanceArg)
+        Tag(""BeforeStatic"", StaticArg)
+        Tag(""BeforeExtension"", ExtensionArg)
+
+        InstanceArg.InstanceMethod()
+        StaticArg.StaticMethod()
+        ExtensionArg.ExtensionMethod()
+
+        Tag(""AfterInstance"", InstanceArg)
+        Tag(""AfterStatic"", StaticArg)
+        Tag(""AfterExtension"", ExtensionArg)
+    End Sub
+
+    Private Sub InstanceMethod()
+    End Sub
+
+    Private Shared Sub StaticMethod()
+    End Sub
+
+    Private Shared Sub Tag(Name As String, Arg As Object)
+    End Sub
+
+End Class
+
+Public Module Extensions
+
+    <Runtime.CompilerServices.Extension>
+    Public Sub ExtensionMethod(S As Sample)
+    End Sub
+
+End Module";
+            var validator = new SETestContext(code, AnalyzerLanguage.VisualBasic, Array.Empty<SymbolicCheck>()).Validator;
+            validator.ValidateContainsOperation(OperationKind.ObjectCreation);
+            validator.ValidateTag("BeforeInstance", x => x.Should().BeNull());
+            validator.ValidateTag("BeforeStatic", x => x.Should().BeNull());
+            validator.ValidateTag("BeforeExtension", x => x.Should().BeNull());
+            validator.ValidateTag("AfterInstance", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue("Instance method should set NotNull constraint."));
+            validator.ValidateTag("AfterStatic", x => x.Should().BeNull("Static method can execute from null instances."));
+            validator.ValidateTag("AfterExtension", x => x.Should().BeNull("Extensions can run on null instances."));
+        }
+
+        [DataTestMethod]
+        [DataRow("DoSomething();")]
+        [DataRow("this.DoSomething();")]
+        [DataRow("(this).DoSomething();")]
+        [DataRow("(((this))).DoSomething();")]
+        public void InstanceMethodCallNotClearsFieldOnThis(string invocation)
```

Comment:
Prefix because of partial class and slight rename
```suggestion
        public void Invocation_SInstanceMethodCallDoesNotClearFieldOnThis(string invocation)
```